File: src/main/java/ukalus/ui/jcurses/PlanBook.java
package ukalus.ui.jcurses;

import ukalus.persistence.Persistence;
import ukalus.persistence.PersistenceException;

import java.io.Serializable;

/**
 * DOCUMENT ME!
 * 
 * @author pmgremo
 */
public class PlanBook implements Serializable {

  private static final long serialVersionUID = 3690195434467308848L;
  private String name;
  private Object page;

  /**
   * DOCUMENT ME!
   * 
   * @param key
   *          DOCUMENT ME!
   */
  public void turnTo(String key) {
    if ((name != null) && !name.equals(key)) {
      try {
        Persistence.put(name, page);
      } catch (PersistenceException e) {
        e.printStackTrace();
      }
    }

    name = key;

    try {
      page = Persistence.get(name);
    } catch (PersistenceException e) {
      e.printStackTrace();
    }
  }

  /**
   * DOCUMENT ME!
   * 
   * @return DOCUMENT ME!
   */
  public Object get() {
    return page;
  }

  /**
   * DOCUMENT ME!
   * 
   * @param key
   *          DOCUMENT ME!
   * @param page
   *          DOCUMENT ME!
   */
  public void add(String key, Object page) {
    turnTo(key);
    this.page = page;
  }
}

File: src/main/java/ukalus/ui/jcurses/PlanBook.kt
package ukalus.ui.jcurses

import ukalus.persistence.Persistence
import ukalus.persistence.PersistenceException

import java.io.Serializable

/**
 * DOCUMENT ME!

 * @author pmgremo
 */
class PlanBook : Serializable {
    private var name: String? = null
    private var page: Any? = null

    /**
     * DOCUMENT ME!

     * @param key
     * *          DOCUMENT ME!
     */
    fun turnTo(key: String) {
        if (name != null && name != key) {
            try {
                Persistence.put(name!!, page!!)
            } catch (e: PersistenceException) {
                e.printStackTrace()
            }

        }

        name = key

        try {
            page = Persistence[name!!]
        } catch (e: PersistenceException) {
            e.printStackTrace()
        }

    }

    /**
     * DOCUMENT ME!

     * @return DOCUMENT ME!
     */
    fun get(): Any? {
        return page
    }

    /**
     * DOCUMENT ME!

     * @param key
     * *          DOCUMENT ME!
     * *
     * @param page
     * *          DOCUMENT ME!
     */
    fun add(key: String, page: Any) {
        turnTo(key)
        this.page = page
    }

    companion object {

        private const val serialVersionUID = 3690195434467308848L
    }
}